nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -387,6 +390,28 @@ def __init__(
     ) -> None:
         super().__init__(precision, device, scaler=scaler)
 
+    def optimizer_step(
```

Comment:
Why is this being added here?

Reply:
This is almost identical to the [parent class impl](https://github.com/Lightning-AI/lightning/blob/6a9d626e189038f09a0cf7157f61e75300ea99cc/src/pytorch_lightning/plugins/precision/native_amp.py#L70). The major difference is that it doesn't manually unscale the gradients. The distributed optimizer plugs into a custom interface in `GradScaler` and I previously ran into problems when calling `GradScaler.unscale_`.

That said, I'll remove it. megatron_amp_o2 only supports bf16, which doesn't do grad scaling.